objfile/patterns.go

```diff
@@ -0,0 +1,281 @@
+package objfile
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+
+	"rsc.io/binaryregexp"
+)
+
+func contains(s []rune, c rune) bool {
+	for _, v := range s {
+		if v == c {
+			return true
+		}
+	}
+
+	return false
+}
+
+func isHexRune(c rune) bool {
+	return contains([]rune{'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'}, c)
+}
+
+func isHex(s string) bool {
+	for _, c := range s {
+		if !isHexRune(c) {
+			return false
+		}
+	}
+	return true
+}
+
+// translate from a yara-style pattern, like:
+//
+//	{ 48 8D 0? ?? ?? ?? ?? EB ?? 48 8? 8? ?? 02 00 00 66 0F 1F 44 00 00 }
+//
+// to a regular expression string compatible with the binaryregexp module, like:
+//
+//	\x48\x8D[\x00-\x0F]....\xEB.\x48[\x80-\x8F][\x80-\x8F].\x02\x00\x00\x66\x0F\x1F\x44\x00\x00
+//
+// although this requires more code, we provide this functionality
+// because these patterns are *much* more readable than raw regular expressions,
+// we strongly value people being able to understand GoReSym's algorithm.
+func RegexpPatternFromYaraPattern(pattern string) (string, error) {
```

Comment:
This is some non-trivial work, is there a significant speedup if we hardcoded the regex style patterns rather than converting at runtime? Also: coooool

Reply:
I agree it's (probably) not a hot path worth optimizing, I was just curious on the effect it would have.